docs/IE_DG/DynamicBatching.md

```diff
@@ -1,52 +1,106 @@
+
+
+You can activate Dynamic Batching by setting `KEY_DYN_BATCH_ENABLED` flag to `YES` in a configuration map that is
 passed to the plugin while loading a network.
-This configuration creates an <code>ExecutableNetwork</code> object that will allow setting batch size
-dynamically in all of its infer requests using <code>SetBatch()</code> method.
-The batch size that was set in passed <code>CNNNetwork</code> object will be used as a maximum batch size limit.
+This configuration creates an `ExecutableNetwork` object that will allow setting batch size
+dynamically in all of its infer requests using `SetBatch()` method.
+The batch size that was set in the passed `CNNNetwork` object will be used as a maximum batch size limit.
 
 Here is a code example:
 
 @snippet snippets/DynamicBatching.cpp part0
 
 
-## Limitations
+### Limitations
 
-Currently, certain limitations for using Dynamic Batching exist:
+Currently, there are certain limitations for the use of Dynamic Batching exist:
 
 * Use Dynamic Batching with CPU and GPU plugins only.
+* Use Dynamic Batching on topologies that consist of certain layers only:
+   * Convolution
+   * Deconvolution
+   * Activation
+   * LRN
+   * Pooling
+   * FullyConnected
+   * SoftMax
+   * Split
+   * Concatenation
+   * Power
+   * Eltwise
+   * Crop
+   * BatchNormalization
+   * Copy
+
+The following types of layers are not supported:
+
+* Layers that might arbitrary change tensor shape (such as Flatten, Permute, Reshape)
+* Layers specific to object detection topologies (ROIPooling, ProirBox, DetectionOutput)
+* Custom layers
+
+Topology analysis is performed during the process of loading a network into plugin, and if the topology is not supported, an exception is generated.
+
+## Using Dynamic Batching (Python)
+
+@sphinxdirective
+.. raw:: html
+
+    <div id="switcher-python" class="switcher-anchor">Python</div>
+@endsphinxdirective
 
+Dynamic Batching is a feature that allows you to dynamically change batch size for inference calls within a preset batch size limit. This feature might be useful when batch size is unknown beforehand, and using extra large batch size is not desired or impossible due to resource limitations. For example, face detection with person age, gender, or mood recognition is a typical usage scenario.
+
+You can activate Dynamic Batching by setting the "DYN_BATCH_ENABLED" flag to "YES" in a configuration map that is passed to the plugin while loading a network. This configuration creates an `ExecutableNetwork` object that will allow setting batch size dynamically in all of its infer requests using the  [ie_api.batch_size](api/ie_python_api/_autosummary/openvino.inference_engine.IENetwork.html#openvino.inference_engine.IENetwork.batch_size) method. The batch size that was set in the passed CNNNetwork object will be used as a maximum batch size limit.
+
+```python
```

Comment:
This example is incomplete. It shows how to load the model with dynamic batching enabled, but not how to do inference on such a model with a dynamic batch size. It is not intuitive how this works, so an example that shows how to propagate a batch with 5 elements through a network with dynamic batching with max size of 32 would be useful. @jiwaszki can you help with this?